### PR TITLE
ADR - replace fetch mocking with msw 

### PR DESCRIPTION
**Pull request by aditudorache**
This PR is a ADR change proposal for using the popular msw for mocking the fetch request in the unit tests.

### CODE REVIEW

File: docs/adr/0010-stop-mocking-fetch-in-tests.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Technical description
+The `msw` server intercepts the fetch calls and returns the configured response. This is done outside the test file and only overrides of the standard responses from the default configuration need to be added to the respective test file. [Here is the documentation](https://github.com/mswjs/msw).
+
+## Consideratons
+Here are few advantages of introducing this library:
+- All the fetch/mocking can moved to one place (internals/testing)
+- The tests involving fetch will be simpler and easier to read/write
+- Enforces using the correct data/types, no half mock objects
+
```
Comment (janjaap) [inline docs/adr/0010-stop-mocking-fetch-in-tests.md:21]: Plus, the e2e tests can also benefit from the msw setup and can use fixtures, the same way the unit tests can.

**Review by boris-arkenaar — APPROVED** [review]

File: docs/adr/0010-stop-mocking-fetch-in-tests.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Context
+
+For mocking unittests we are using at this moment the `jest-fetch-mock` package. This works but there are new develoments on the market. As the creator of the [Testing Library](https://testing-library.com/) suggests in this [article](https://kentcdodds.com/blog/stop-mocking-fetch), there is a new strategy available that would simplify testing of components that are making fetch calls.
+
+## Technical description
+The `msw` server intercepts the fetch calls and returns the configured response. This is done outside the test file and only overrides of the standard responses from the default configuration need to be added to the respective test file. [Here is the documentation](https://github.com/mswjs/msw).
+
+## Consideratons
+Here are few advantages of introducing this library:
+- All the fetch/mocking can moved to one place (internals/testing)
```
Comment (MauritsR) [inline docs/adr/0010-stop-mocking-fetch-in-tests.md:18]: What would this look like in tests that require different responses to the same call? Magic input that returns a specific response?

Comment (MauritsR) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:18]: Random example I found: 

`fetch.mockResponses([JSON.stringify({ detail: 'filled out' }), { status: 410 }]);`

This is already pretty succinct - how would this be changed with `msw`?

File: docs/adr/0010-stop-mocking-fetch-in-tests.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- The tests involving fetch will be simpler and easier to read/write
+- Enforces using the correct data/types, no half mock objects
```
Comment (MauritsR) [inline docs/adr/0010-stop-mocking-fetch-in-tests.md:20]: Typing the mocks, does that mean the fixtures itself will become .ts files?

File: docs/adr/0010-stop-mocking-fetch-in-tests.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Refactoring plan
+- Add the `msw` package and create the configuration in the `internals folder`
+- Use for the configuration the existing features that are already present in `utils/__tests__/fixtures`
+- Start small, replace the `jest-fetch-mock` only in the changed files.
+
+
+## Consequences
+
+- There will be a transition period where both `jest-fetch-mock` and `msw` will coexist.
```
Comment (MauritsR) [inline docs/adr/0010-stop-mocking-fetch-in-tests.md:35]: I'm a bit worried about adding yet another tool to run alongside a similar one. 

We're already in a transition phase for 
- JavaScript -> TypeScript
- naming conventions
- form handling
- usage of hooks/redux

If we have a running example, we might be able to see if we can implement it in the entire app at once

File: docs/adr/0010-stop-mocking-fetch-in-tests.md (hunk 2 above)
Comment (janjaap) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:18]: See https://kentcdodds.com/blog/stop-mocking-fetch

Comment (aditudorache) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:18]: Will workout a poc asap so we solve this kind of scenarios

File: docs/adr/0010-stop-mocking-fetch-in-tests.md (hunk 3 above)
Comment (aditudorache) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:20]: No, they will stay .json but can be typechecked.

File: docs/adr/0010-stop-mocking-fetch-in-tests.md (hunk 4 above)
Comment (aditudorache) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:35]: At atlas, they have already solved some of the challenges we might have. The aim is to migrate gradually as files are touched, will have to set it up though.

File: docs/adr/0010-stop-mocking-fetch-in-tests.md (hunk 2 above)
Comment (MauritsR) [reply docs/adr/0010-stop-mocking-fetch-in-tests.md:18]: Looking forward to it! Would like to doublecheck the advantages list and see how this list holds up (and if we cannot simply do it with the current solution as well)

**Comment by aditudorache** [comment]
Although it looks promising, there are some issues I couldn't yet solve. There seem to be some async issues when running the tests in our project. Please see here the progress I made: https://github.com/Amsterdam/signals-frontend/tree/SIG-3485-add-msw-for-testing.
We might consider drop this proposal when no easy solution can be found soon.

**Review by MauritsR — APPROVED** [review]

**Pull request closed without merging**